pipelines/skylab/build_indices/BuildIndices.wdl

```diff
@@ -106,365 +26,70 @@ task BuildStarSingleNucleus {
-
-    # Compressed fasta required here
-    gzip ~{references.genome_fa}
-
-    # download snp file
-    wget http://hgdownload.cse.ucsc.edu/goldenPath/~{genome_short_string}/database/~{snp_file}.gz
-    gunzip ~{snp_file}.gz
-
-    # extract snps, splice sites, and exon information
-    $HISAT2_DIR/hisat2_extract_snps_UCSC.py ~{references.genome_fa}.gz ~{snp_file} genome
-    $HISAT2_DIR/hisat2_extract_splice_sites.py ~{references.annotation_gtf} > genome.ss
-    $HISAT2_DIR/hisat2_extract_exons.py ~{references.annotation_gtf} > genome.exon
-
-    # build the hisat2 reference
-    $HISAT2_DIR/hisat2-build \
-      -p 8 \
-      genome.fa \
-      --snp genome.snp \
-      --haplotype genome.haplotype \
-      --ss genome.ss \
-      --exon genome.exon \
-      genome_snp_tran
-
-    mkdir ~{ref_name}
-    cp ./*.ht2 ~{ref_name}
-    tar -zcvf ~{hisat2_index_name} ~{ref_name}
-
-  >>>
-
-  runtime {
-    docker:"quay.io/humancellatlas/secondary-analysis-hisat2:v0.3.0-2-2.1.0"
-    memory: "240 GiB"
-    disks: "local-disk 100 HDD"
-    cpu: "16"
-  }
-  output {
-    File hisat2_index = hisat2_index_name
-  }
-}
-
-task BuildPicardRefFlat {
-  input {
-    References references
-  }
-
-  String refflat_name = basename(references.annotation_gtf, ".gtf") + ".refflat.txt"
-
-  command {
-    set -eo pipefail
-
-    gtfToGenePred -genePredExt -geneNameAsName2  ~{references.annotation_gtf} refflat.tmp.txt
-
-    paste <(cut -f 12 refflat.tmp.txt) <(cut -f 1-10 refflat.tmp.txt) > ~{refflat_name}
-
-  }
-
-  runtime {
-    docker: "quay.io/humancellatlas/gtf_to_genepred:v0.0.0"
-    memory: "8 GiB"
-    disks: "local-disk 100 HDD"
-    cpu: "8"
-  }
-
-  output {
-      File refflat = refflat_name
-  }
-}
-
-task BuildIntervalList {
-  input {
-    References references
-  }
-
-  String interval_list_name = basename(references.annotation_gtf, ".gtf") + ".interval_list"
-
-  command <<<
-    set -eo pipefail
-
-
-    # index the fasta file
-
-    samtools faidx ~{references.genome_fa}
-    cut -f1,2 ~{references.genome_fa}.fai > sizes.genome
-
-    awk -F '\t'  '{  printf "@SQ\tSN:%s\tLN:%s\n", $1, $2 }' sizes.genome  >> ~{interval_list_name}
-
-    grep 'gene_type "rRNA"' ~{references.annotation_gtf} |
-        awk '$3 == "transcript"' |
-    cut -f1,4,5,7,9 |
-    perl -lane '
-        /transcript_id "([^"]+)"/ or die "no transcript_id on $.";
-        print join "\t", (@F[0,1,2,3], $1)
-    ' |
-    sort -k1V -k2n -k3n  >> ~{interval_list_name}
-
-  >>>
-
-  runtime {
-    docker: "quay.io/humancellatlas/secondary-analysis-umitools:0.0.1"
-    memory: "8 GiB"
-    disks: "local-disk 100 HDD"
-    cpu: "8"
-  }
-
-  output {
-      File interval_list = interval_list_name
-  }
-}
 
 
 workflow BuildIndices {
   input {
-    String gtf_version
-    String organism
-    String organism_prefix
-    String genome_short_string
-    String dbsnp_version
+    File annotations_gtf
+    File genome_fa
+    String gtf_version = "M31"
```

Comment:
Should we have a default value here? If yes, perhaps all the others should also be defaulted?

Reply:
removed the default